### ISSUE TITLE
static_host not requesting https

### ISSUE THREAD

**Issue by BrendanMartin** [pallets/flask#3024]
I have a Flask app on Google App Engine and my static files are in a Google bucket.

I'm using `static_host` like so:

```python
def create_app(config_name):
    app = Flask(__name__, host_matching=True, static_host='storage.googleapis.com/mybucket')
```
And I'm also using Flask-SSLify for the basic redirect from http to https:

```python
app = create_app(os.getenv('FLASK_CONFIG') or 'default')
sslify = SSLify(app, permanent=True)
```

But when I launch on App Engine my static files won't load because they are being requested with http. In other words, my app is successfully on https but the static file URLs are being built like `http://storage.googleapis.com/mybucket/...` without https.

It seems SSLify is just building a `before_request` to do redirects, so I'm not sure where to tell Flask to request my static host with https.

**Comment by davidism** [comment]
This is not what `static_host` is for. `static_host` is for when you have multiple host names assigned to your server, and you want to route based on which host the request is made to. `storage.googleapis.com` is presumably not routing to your application.

**Comment by BrendanMartin** [comment]
I do have multiple host names. Currently I have example1.com and example2.com pointing to the same app. I need to have the static files served from a Google bucket.

**Comment by davidism** [comment]
Right, but that's not the configuration that `static_host` is for. `static_host` is when you still want *Flask* to serve your static files, it tells it what host that's coming in to Flask to use for that.

You'll want to configure App Engine to serve requests to `/static` from your bucket rather than passing to Flask, just like you'd configure a server like Nginx.

**Comment by davidism** [comment]
Also, what does `url_for('some other endpoint', _external=True)` return? If it's not HTTPS either, there's a misconfiguration, either in SSLify or `app.config`. I'm not familiar with SSLify or App Engine though.

**Comment by BrendanMartin** [comment]
Adding `_external=True` actually did fix it.

Here's what I used in my base template:
`<link rel="stylesheet" media="all" href="{{ url_for('static', filename='style.css',  _external=True) }}"/>`

Out:
`https://storage.googleapis.com/bucket/static/style.css`

But I also added this too:
`    app.config.update(dict(
            PREFERRED_URL_SCHEME='https'
    ))`
